scripts/stubsabot.py

```diff
@@ -281,7 +289,118 @@ async def get_diff_url(
     async with session.get(diff_url, headers=get_github_api_headers()) as response:
         # Double-check we're returning a valid URL here
         response.raise_for_status()
-    return diff_url
+    return GithubDiffInfo(repo_path=github_info.repo_path, old_tag=old_tag, new_tag=new_tag, diff_url=diff_url)
+
+
+FileInfo: TypeAlias = dict[str, Any]
+
+
+def _plural_s(num: int, /) -> str:
+    return "s" if num != 1 else ""
+
+
+@dataclass
+class DiffAnalysis:
+    py_files: list[FileInfo]
+    py_files_stubbed_in_typeshed: list[FileInfo]
+
+    @property
+    def runtime_definitely_has_consistent_directory_structure_with_typeshed(self) -> bool:
+        """
+        If 0 .py files in the GitHub diff exist in typeshed's stubs,
+        there's a possibility that the .py files might be found
+        in a different directory at runtime.
+
+        For example: pyopenssl has its .py files in the `src/OpenSSL/` directory at runtime,
+        but in typeshed the stubs are in the `OpenSSL/` directory.
+        """
+        return bool(self.py_files_stubbed_in_typeshed)
+
+    @functools.cached_property
+    def public_files_added(self) -> Sequence[str]:
+        return [
+            file["filename"]
+            for file in self.py_files
+            if not re.match("_[^_]", Path(file["filename"]).name) and file["status"] == "added"
+        ]
+
+    @functools.cached_property
+    def typeshed_files_deleted(self) -> Sequence[str]:
+        return [file["filename"] for file in self.py_files_stubbed_in_typeshed if file["status"] == "removed"]
+
+    @functools.cached_property
+    def typeshed_files_modified(self) -> Sequence[str]:
+        return [file["filename"] for file in self.py_files_stubbed_in_typeshed if file["status"] in {"modified", "renamed"}]
+
+    @property
+    def total_lines_added(self) -> int:
+        return sum(file["additions"] for file in self.py_files)
+
+    @property
+    def total_lines_deleted(self) -> int:
+        return sum(file["deletions"] for file in self.py_files)
+
+    @staticmethod
+    def _describe_files(*, verb: str, filenames: Sequence[str]) -> str:
+        num_files = len(filenames)
+        if num_files > 1:
+            description = f"have been {verb}"
+            # Don't list the filenames if there are *loads* of files
+            if num_files <= 5:
```

Comment:
Maybe make this a global constant? Also we could probably make it a bit higher than 5 before it becomes overwhelming.

Reply:
I made it a `ClassVar` constant, and bumped it up to 7.